lib/assert.js

```diff
@@ -61,11 +61,7 @@ assert.AssertionError = function AssertionError(options) {
 util.inherits(assert.AssertionError, Error);
 
 function truncate(s, n) {
-  if (typeof s === 'string') {
-    return s.length < n ? s : s.slice(0, n);
-  } else {
-    return s;
-  }
+  return s.length < n ? s : s.slice(0, n);
```

Comment:
Why not directly `s.slice(0, n)`? The `String.prototype.slice` seems to cover the case `s.length < endSlice` and returned what we expected.